gitpod-io/gitpod
# Update Helm dependencies in Installer

## Description

This updates all Helm dependencies in the Installer (Docker registry, Minio, MySQL, RabbitMQ) to the latest versions. This PR has more commits than is strictly necessary, but have done it like this to ensure that the changes to the golden files are human-readable.

All of these dependency upgrade change the major version of the Helm chart. The Bitnami charts change how the image block works, so the previous way of defining the custom pull registry no longer works - in retrospect, I wish I'd done the original definition in the way I've done it here. The first two commits ([1](https://github.com/gitpod-io/gitpod/pull/15568/commits/0a9c537b854b971017c67b59e23be02720b63c9e) and [2](https://github.com/gitpod-io/gitpod/pull/15568/commits/853b6fd9a0835cf2bbdced5d0d81032fb6fb3986)) address this by adding a new golden file to handle a custom pull repo and change the image block in the Helm dependencies.

The [third commit](https://github.com/gitpod-io/gitpod/pull/15568/commits/09068b829fe4fc2d916ac6e6290d2c32cb820c2d) exists because the Docker registry [introduced a checksum on the secret.yaml](https://github.com/twuni/docker-registry.helm/pull/49) in the template. This uses the Helm `randAlphaNum` function, which breaks the random seed - 64 bits wasn't enough random data, so increased to 128 bits.

The [fourth commit](https://github.com/gitpod-io/gitpod/pull/15568/commits/631f0e54cb22eb6296c1b20e7237655271211ef2) is the actual update of the Helm dependencies and the associated golden files.

Finally, there is an update made to `go-gorp` as warning "Versions prior to 3.0.4 had a vulnerability in the dependency graph" reported (this could be a separate PR, but it's all part of the same area so I've put it in here).

## Related Issue(s)

Fixes https://github.com/gitpod-io/security/issues/95

## How to test

Use [preview environment](https://sje-instal6626b9bf74.preview.gitpod-dev.com/workspaces)/self-hosted instance and do normal smoke-tests:
- build image
- start workspace
- use `gp stop`
- ensure that backups are saved to storage

## Release Notes

```release-note
Update Helm dependencies in Installer
```

## Documentation


## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
